GVFS/GVFS.Platform.Windows/WindowsFileSystem.cs

```diff
@@ -117,6 +117,58 @@ public bool IsSocket(string fileName)
             return false;
         }
 
+        public void CreateDirectoryAccessibleByAuthUsers(string directoryPath)
+        {
+            if (Directory.Exists(directoryPath))

```

Comment:
Do we want to emit a warning if the directory has already been created but does _not_ have the correct ACLs?

Reply:
We'll definitely want to do something (either warn the user or fix up the ACLs on the directory).

My goal was to scope the changes in this PR so that:

- We improve the situation on clean machines
- We don't make things any worse on machines with existing repos/cache directories

I think your suggestion falls into the work required for #1580.